src/java/com/google/tagpost/spanner/SpannerService.java

```diff
@@ -0,0 +1,90 @@
+package com.google.tagpost.spanner;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.*;
+import com.google.tagpost.Comment;
+import com.google.tagpost.Tag;
+import com.google.tagpost.Thread;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Data access and operations of Spanner */
+public class SpannerService implements DataService {
+
+  @Override
+  public List<Thread> getAllThreadsByTag(Tag tag) {
+
+    SpannerOptions spannerOptions = SpannerOptions.newBuilder().build();
+    Spanner spanner = spannerOptions.getService();
+    List<Thread> threadList = new ArrayList<>();
+
+    try {
+      DatabaseId db = DatabaseId.of("testing-bigtest", "tagpost", "test");
+      DatabaseClient dbClient = spanner.getDatabaseClient(db);
+      String query =
+          "SELECT ThreadID, PrimaryTag FROM Thread WHERE PrimaryTag = '" + tag.getTagName() + "'";
```

Comment:
Don't construct SQL queries with variables. Use [Statement.Builder](https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/Statement.html#newBuilder-java.lang.String-) and [parameters](https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/Statement.Builder.html#bind-java.lang.String-).

Reply:
Fixed .